src/common/AppToggle.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <label class="container">
+    <input
+      v-bind="$attrs"
+      class="input"
+      type="checkbox"
+      :checked="modelValue"
+      @change="$emit('update:modelValue', $event.target.checked)"
+    />
+
+    <span class="switch">
+      <span :class="['trigger', { disabled: modelValue }]">{{ leftText }}</span>
+      <span :class="['trigger', { disabled: !modelValue }]">{{
+        rightText
+      }}</span>
+    </span>
+  </label>
+</template>
+
+<script>
+export default {
+  props: {
+    leftText: {
+      type: String,
+      required: true
+    },
+    rightText: {
+      type: String,
+      required: true
+    },
+    modelValue: {
+      type: Boolean,
+      required: true
+    }
+  },
+  emits: ["update:modelValue"]
+};
+</script>
+
+<style scoped>
+.container {
+  cursor: pointer;
+  display: flex;
+  box-sizing: border-box;
+  min-width: 20rem;
```

Comment:
I added a `min-width` so the text doesn't look squished — the element can still grow larger with longer texts.